plaidml/bridge/openvino/ops/deformable_convolution.cpp

```diff
@@ -0,0 +1,290 @@
+  deformed_index_reshape_dims.push_back(1);
+  auto deformed_index_slice = op::slice(deformed_index).add_dim(0, N).add_dim(0, CI);
+  for (auto i = 0; i < rank - 2; ++i) {
+    deformed_index_slice = op::slice(deformed_index_slice).add_dim(0, NEW_DIM[i]);
+  }
+  for (auto i = 0; i < rank - 2; ++i) {
+    deformed_index_vec[i] = op::slice(deformed_index_slice).add_dim(i);
+    deformed_index_vec_ceil[i] = edsl::ceil(deformed_index_vec[i]);
+    deformed_index_vec_floor[i] = edsl::floor(deformed_index_vec[i]);
+    deformed_index_vec_ceil[i] = op::reshape(deformed_index_vec_ceil[i], make_tuple<int>(deformed_index_reshape_dims));
+    deformed_index_vec_floor[i] =
+        op::reshape(deformed_index_vec_floor[i], make_tuple<int>(deformed_index_reshape_dims));
+  }
+  // Coord_num is the number of tensor required for multi-linear interpolation.
+  int coord_num = static_cast<int>(std::pow(2.0, rank - 2));
+  std::vector<edsl::Tensor> deformed_index_coord_vec(coord_num);
+  std::vector<edsl::Tensor> deformed_input_vec(coord_num);
+  for (auto i = 0; i < coord_num; ++i) {
+    std::vector<edsl::Tensor> concat_vec(rank - 2);
+    for (auto j = 0; j < rank - 2; ++j) {
+      int flag = (i / static_cast<int>(std::pow(2.0, rank - 2 - j - 1))) % 2;
+      concat_vec[j] = (flag == 0) ? deformed_index_vec_floor[j] : deformed_index_vec_ceil[j];
+    }
+    deformed_index_coord_vec[i] = op::concatenate(concat_vec, rank);
+    deformed_index_coord_vec[i] = edsl::cast(deformed_index_coord_vec[i], DType::INT32);
+    deformed_input_vec[i] = edsl::gather(I, deformed_index_coord_vec[i]).mode(edsl::GatherMode::ND).batchDims(2);
+  }
+  // Change the shape of deformed_index_ceil tensor and deformed_index_floor tensor to original shape.
+  deformed_index_reshape_dims.erase(deformed_index_reshape_dims.end() - 1);
+  for (auto i = 0; i < rank - 2; ++i) {
+    deformed_index_vec_ceil[i] = op::reshape(deformed_index_vec_ceil[i], make_tuple<int>(deformed_index_reshape_dims));
+    deformed_index_vec_floor[i] =
+        op::reshape(deformed_index_vec_floor[i], make_tuple<int>(deformed_index_reshape_dims));
+  }
+  // Get deformed input tensor with multi-linear interpolation.
+  for (auto i = 0; i < rank - 2; ++i) {
+    for (auto j = 0; j < static_cast<int>(std::pow(2.0, rank - 2 - i - 1)); ++j) {
+      deformed_input_vec[j] = deformed_input_vec[j] * (deformed_index_vec_ceil[i] - deformed_index_vec[i]) +
+                              deformed_input_vec[j + static_cast<int>(std::pow(2.0, rank - 2 - i - 1))] *
+                                  (1.0 - deformed_index_vec_ceil[i] + deformed_index_vec[i]);
+    }
+  }
+  edsl::Tensor deformed_input = deformed_input_vec[0];
+  // Set up the new strides.
+  std::vector<int> new_strides(F_shape.begin() + 2, F_shape.end());
+  // Compute DeformableConvolution.
+  edsl::Tensor result = op::convolution(deformed_input, F)
+                            .strides(new_strides)
```

Comment:
you can use `std::vector<int>{F_shape.begin() + 2, F_shape.end()}` and delete line 208,209